index.js

```diff
@@ -157,6 +152,92 @@ app.get('/logout', (req, res) => {
     }
 });
 
+app.get('/dashboard', async (req, res) => {
+    if (!req.session.user || !req.cookies.user_sid) {
+        res.redirect('/login')
+        return
+    }
+
+    try {
+        let allLocations = await db.getLocations()
+        
+        let viewParams = {
+            locations: allLocations
+                .map(location => { 
+                    return { name: location.displayName, id: location.locationid }
+                })
+        }
+        viewParams.viewMessage = allLocations.length >= 1 ? 'Please select a location' : 'No locations to display'
+
+        res.send(Mustache.render(locationsDashboardTemplate, viewParams))
+    }
+    catch(err) {
+        console.log(err)
+        res.status(500).send()
+    }
+})
+
+app.get('/dashboard/:locationId', async (req, res) => {
+    if (!req.session.user || !req.cookies.user_sid) {
+        res.redirect('/login')
+        return
+    }
+    else if(typeof req.params.locationId !== "string") {
+        let locations = await db.getlocations()
+        res.redirect('/dashboard/' + locations[0].id)
+        return
+    }
+
+    try {
+        let recentSessions = await db.getHistoryOfSessions(req.params.locationId)
+        let currentLocation = await db.getLocationData(req.params.locationId)
+        let allLocations = await db.getLocations()
+        
+        let viewParams = {
+            recentSessions: [],
+            currentLocationName: currentLocation.display_name,
+            locations: allLocations
+                .map(location => { 
+                    return { name: location.displayName, id: location.locationid }
+                })
+        }
+
+        // commented out keys were not shown on the old frontend but have been included in case that changes
+        for(const recentSession of recentSessions) {
+            let startTime = moment(recentSession.startTime, moment.ISO_8601)
+                .tz('America/Vancouver')
+                .format('DD MMM Y, hh:mm:ss A')
+            let endTime = recentSession.endTime 
+                ? moment(recentSession.endTime, moment.ISO_8601)
+                    .tz('America/Vancouver')
+                    .format('DD MMM Y, hh:mm:ss A')
+                : 'Ongoing'
+
```

Comment:
I was not sure about how best to format this but didn't want to have a conditional in the `viewParams` object, or a massive one-liner, so feedback would be appreciated!

Reply:
does the job!

Reply:
The only thing I would say here is that moment is pretty much deprecated and we should move to something different like date fns or even just the bog standard native date library.

But this is ok for now and not the biggest deal

Reply:
for details see: https://momentjs.com/docs/#/-project-status/

Reply:
ooh ok that's good to know thanks!

Reply:
could maybe be another task, where we go through and remove all moment occurences, any thoughts @trisapeace ?

Reply:
The `? :` expression you have seems fine to me.

If you want to make a little cleaner, you could factor out the time formatting part into its own function. It would also help us if we ever decide to change the format then we'd only have to do it in one place and it would continue to be consistent in the two columns. Up to you though, it's not a big deal.

Another task sounds like a good idea @mariocimet .

Reply:
I think it would make sense to do that as part of the task of removing moment maybe if we're going that route?